agent/src/health.rs

```diff
@@ -0,0 +1,396 @@
+// This file is @generated by prost-build.
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct MetricRequest {
+    #[prost(string, tag = "1")]
+    pub name: ::prost::alloc::string::String,
+}
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct MetricReply {
+    #[prost(string, tag = "1")]
+    pub host_name: ::prost::alloc::string::String,
+    #[prost(string, tag = "2")]
+    pub name: ::prost::alloc::string::String,
+    #[prost(string, tag = "3")]
+    pub kernel_version: ::prost::alloc::string::String,
+    #[prost(string, tag = "4")]
+    pub os_version: ::prost::alloc::string::String,
+    #[prost(string, tag = "5")]
+    pub memory: ::prost::alloc::string::String,
+    #[prost(string, tag = "6")]
+    pub used_memory: ::prost::alloc::string::String,
+    #[prost(string, tag = "7")]
+    pub percent_memory: ::prost::alloc::string::String,
+    #[prost(string, tag = "8")]
+    pub cpu: ::prost::alloc::string::String,
+}
+#[allow(clippy::derive_partial_eq_without_eq)]
+#[derive(Clone, PartialEq, ::prost::Message)]
+pub struct Metric {
+    #[prost(string, tag = "1")]
+    pub host_name: ::prost::alloc::string::String,
+    #[prost(string, tag = "2")]
+    pub name: ::prost::alloc::string::String,
+    #[prost(string, tag = "3")]
+    pub kernel_version: ::prost::alloc::string::String,
+    #[prost(string, tag = "4")]
+    pub os_version: ::prost::alloc::string::String,
+    #[prost(string, tag = "5")]
+    pub memory: ::prost::alloc::string::String,
+    #[prost(string, tag = "6")]
+    pub used_memory: ::prost::alloc::string::String,
+    #[prost(string, tag = "7")]
+    pub percent_memory: ::prost::alloc::string::String,
+    #[prost(string, tag = "8")]
+    pub cpu: ::prost::alloc::string::String,
+}
```

Comment:
I believe these structures are already made available by importing your protos with Tonic ;

Tonic takes care of pre-generating the code defined in the proto (and some associated methods),
so my understanding is that it's not needed to redefine them here ; nor to make them public (since they can be accessed everywhere the protos are imported)

Reply:
Yes we fought throught to find the right technique to import those file. I found the right method yesterday but i didn't shared it with the team yet. This file will be removed in the end

Reply:
Do you think we should agree on our techniques and share our methods? Or just work separately?
I'd be glad to explain my work if it can help :)